test/test_datapipe.py

```diff
@@ -2073,24 +2073,57 @@ def add_one(self, x):
         def classify(self, x):
             return 0
 
-        def __init__(self):
-            self._dp = dp.iter.IterableWrapper([1, 2, 4]).map(self.add_one).demux(2, self.classify)[0]
+        def __init__(self, fn, source_dp=None):
+            self.fn = fn
+            self.source_dp = source_dp if source_dp else dp.iter.IterableWrapper([1, 2, 4])
+            self._dp = self.source_dp.map(self.add_one).demux(2, self.classify)[0]
```

Comment:
Unrelated to pickle, this is terrible usage, as you should not pass bound methods here. The proper solution would be to have `add_one` declared as `@staticmethod`

Reply:
I agree; I came up with this example to test of serialization would still work under extreme corner cases.

Reply:
I will remove the bound method here